src/main/java/rx/internal/schedulers/SchedulerContextException.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright 2016 Netflix, Inc.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package rx.internal.schedulers;
+
+/**
+ * Used only for providing context around where work was scheduled should an error occur in a different thread.
+ */
+public class SchedulerContextException extends Exception {
+    public static Throwable create() {
+        if (System.getProperty("rxjava.captureSchedulerContext", "false").equals("true")) {
```

Comment:
Probably just a static volatile boolean, right? Atomicity isn't needed
since it's only read by RxJava and only written by application code.

On Fri, May 13, 2016, 11:58 AM David Karnok <notifications@github.com>
wrote:

> In src/main/java/rx/internal/schedulers/SchedulerContextException.java
> <https://github.com/ReactiveX/RxJava/pull/3937#discussion_r63234396>:
>
> > + * http://www.apache.org/licenses/LICENSE-2.0
> > + *
> > + * Unless required by applicable law or agreed to in writing, software
> > + * distributed under the License is distributed on an "AS IS" BASIS,
> > + * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
> > + * See the License for the specific language governing permissions and
> > + * limitations under the License.
> > + */
> > +package rx.internal.schedulers;
> > +
> > +/**
> > + * Used only for providing context around where work was scheduled should an error occur in a different thread.
> > + */
> > +public class SchedulerContextException extends Exception {
> > +    public static Throwable create() {
> > +        if (System.getProperty("rxjava.captureSchedulerContext", "false").equals("true")) {
>
> Okay, in that case, I'd consider a public static AtomicBoolean instead of
> the hash-lookup and string comparison. From security perspective, if you
> can set System properties, you could just as set an AtomicBoolean, right?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/ReactiveX/RxJava/pull/3937/files/9630081dc8ef75edad8dc397a661cf61d4ee57fd#r63234396>
>